text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+# Context Propagation: A Layered Approach 
+
+* [Motivation](#Motivation)
+* [OpenTelemetry layered architecture](#OpenTelemetry-layered-architecture)
+  * [Cross-Cutting Concerns](#Cross-Cutting-Concerns)
+    * [Observability API](#Observability-API)
+    * [Correlations API](#Correlations-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+  * [Global initialization](#Global-initialization)
+  * [Extracting and injecting from HTTP headers](#Extracting-and-injecting-from-HTTP-headers)
+  * [Simplify the API with automated context propagation](#Simplify-the-API-with-automated-context-propagation)
+  * [Implementing a propagator](#Implementing-a-propagator)
+  * [Implementing a concern](#Implementing-a-concern)
+  * [The scope of current context](#The-scope-of-current-context)
+  * [Referencing multiple contexts](#Referencing-multiple-contexts)
+  * [Falling back to explicit contexts](#Falling-back-to-explicit-contexts)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+# Motivation
+
+This RFC addresses the following topics:
+
+**Separation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Correlations) and observability data (SpanContext, etc) 
+  separately.  
+
+**Extensibility**
+* Allow developers to create new applications for context propagation. For 
+  example: A/B testing, encrypted or authenticated data, and new, experimental 
+  forms of observability.
```

Comment:
nit: what are experimental forms of observability? First two examples are very clear and this one is confusing as it's saying that OT is not extensible enough without telling what's missing

Reply:
That's fine, I can remove it. In my head, I was thinking about how we could add additional types of observability without disturbing the existing APIs, and reusing this layer would allow those new APIs to be modeled in an experimental fashion, outside of OTel, before they are added in. But I agree that it is confusing.